### ISSUE TITLE
Project dependencies may have API risk issues

### ISSUE THREAD

**Issue by PyDeps** [sniklaus/3d-ken-burns#65]
Hi, In **3d-ken-burns**, inappropriate dependency versioning constraints can cause risks.

Below are the dependencies and version constraints that the project is using

```
cupy>=5.0.0
flask>=1.0.0
gevent>=1.3.0
moviepy>=1.0.0
numpy>=1.15.0
opencv-contrib-python>=3.4.0
scipy>=1.1.0
torch>=1.6.0
torchvision>=0.7.0
```

The version constraint **==** will introduce the risk of dependency conflicts because the scope of dependencies is too strict.
The version constraint **No Upper Bound** and **\*** will introduce the risk of the missing API Error because the latest version of the dependencies may remove some APIs.

After further analysis, in this project,
The version constraint of dependency **gevent** can be changed to *>=0.10.0,<=21.12.0*.
The version constraint of dependency **moviepy** can be changed to *>=0.2.1.7.15,<=2.0.0.dev2*.
The version constraint of dependency **numpy** can be changed to *>=1.8.0,<=1.23.0rc3*.

The above modification suggestions can reduce the dependency conflicts as much as possible, 
and introduce the latest version as much as possible without calling Error in the projects.

The invocation of the current project includes all the following methods.

<details><summary>The calling methods from the gevent</summary>
 <pre>gevent.sleep
</pre>
</details>
<details><summary>The calling methods from the moviepy</summary>
 <pre>moviepy.editor.ImageSequenceClip
</pre>
</details>
<details><summary>The calling methods from the numpy</summary>
 <pre>numpy.linalg.lstsq
</pre>
</details>
<details><summary>The calling methods from the all methods</summary>
 <pre>glob.glob
numpy.nanmean
intTest.npyImages.transpose
fill_disocclusion.detach
cv2.imdecode
tenExisting.clone.detach
Inpaint.cuda.eval.load_state_dict
tenInput.clone
Inpaint.cuda.eval
moviepy.editor.ImageSequenceClip
npyReference.numpy.log10.npyEstimate.numpy.log10.__abs__
tenMerge.tenMask.clip
flask.Response
Upsample
process_load
torch.nn.Sequential
exec
cv2.imread
fltThr2.append
self.netDisparityThr
self.netImageTwo
objCommon.view
npyReference.npyReference.npyEstimate.__abs__.mean
fltAbsrel.append
pe_ori_wall.tolist
npyReference.npyEstimate.__pow__.mean
disparity_adjustment
super
strTempdir.open.read
netRefine
intTop.intBottom.intTop.round.int.tenPlane.max
flask.Flask
process_shift
tenDisparity.clone.view
torch.nn.functional.threshold
tenPlane.sum.flatten.nonzero.item
numpy.frombuffer
intWidth.intHeight.tenData.shape.tenInput.new_zeros.fill_.nelement
tenOutput.tenDepth.tenInput.preprocess_kernel.launch_kernel
pe_fla_wall.tolist
pe_fla.extend
numpy.sqrt
tenDepth.cpu.numpy
pe_fla_table.tolist
math.floor
tenPlane.sum.flatten.nonzero
round
self.netDisparity.view
numpy.array
gevent.sleep
cv2.resize
valid.objMat.flatten
tenMask.sum.item
Semantics.cuda
cv2.imencode
torch.hub.load_state_dict_from_url.items
self.netShortcut
fill_disocclusion
tenDepth.detach.cpu.numpy
scipy.io.loadmat.flatten.tolist
cupy.memoize
tenDisparity.shape.tenDisparity.view.std
numpy.stack
Disparity
float
numpy.save
self.netContext
fltLogten.append
Semantics
strTensor.objVariables.size
tenDisparity.shape.tenDisparity.view.std.view
tenDisparity.cpu.numpy
self._modules
tenImage.shape.tenImage.view.mean
tenRender.detach.cpu.numpy.transpose.clip
str
objInpainted.max.objInpainted.spatial_filter.abs.float
npyImage.transpose.astype
numpy.full
tempfile.gettempdir
os.makedirs
io.BytesIO
torch.nn.functional.max_pool2d
self.netImage.view
objFile.read
max
numpy.float32.npyImage.transpose.astype.numpy.ascontiguousarray.torch.FloatTensor.cuda
tenInput.clone.data_ptr
self.netDisparity
intMerge.intMerge.objPredictions.float.sum
min
spatial_filter.clone
intV.intU.npyShiftU.item
self.netImage.clip
fltThr3.append
process_shift.view
npyImage.copy.transpose
tenVertical.type_as.type_as
npyReference.numpy.log10.npyEstimate.numpy.log10.__abs__.mean
netInpaint
objFlask.gevent.pywsgi.WSGIServer.serve_forever
torch.cat.data_ptr
intTest.npyDepths.transpose
torchvision.models.detection.maskrcnn_resnet50_fpn
tenMasks.append
arguments_strOut.replace
tenDisparity.max.tenDisparity.spatial_filter.abs.float
tenHorizontal.type_as.type_as
intWidth.intHeight.tenData.shape.tenInput.new_zeros.fill_.data_ptr
self.netImageThr
fltSqrel.append
torch.__version__.split
render_pointcloud
compute_global_errors
cv2.IMWRITE_JPEG_QUALITY.npyKenburns.cv2.imencode.tobytes
torchvision.models.detection.maskrcnn_resnet50_fpn.cuda.eval
h5py.File
flask.jsonify
npyReference.npyEstimate.__pow__.mean.numpy.sqrt.item
intV.intU.npyShiftV.item
tenOutput.detach.clone
enumerate
time.time
strFile.endswith
valid.sum.item
torch.hub.load_state_dict_from_url
tenImage.shape.tenImage.view.std.view
tenExisting.float.sum
tenMerge.tenMask.sum
os.environ.os.environ.tuple.strKernel.cupy.cuda.compile_with_cache.get_function
pe_ori_floor.tolist
torch.nn.Upsample
tenDisparity.max.tenDisparity.spatial_filter.abs
tenInput.data_ptr
Refine.cuda.eval
fltY.fltX.torch.FloatTensor.view
Downsample
numpy.linspace.tolist.index
scipy.io.loadmat.flatten
numpy.maximum
zipfile.ZipFile
tenDepth.shape.tenDepth.shape.tenDepth.shape.torch.linspace.view
objCommon.objCommon.torch.cat.view
Disparity.cuda.eval
numpy.linalg.lstsq
npyReference.npyEstimate.__abs__
objSettings.objSettings.numpy.linspace.repeat
getopt.getopt
intWidth.intHeight.tenData.shape.tenInput.new_zeros.fill_
self.netImage
npyKenburns.reversed.list.npyKenburns.npyFrame.npyFrame.moviepy.editor.ImageSequenceClip.write_videofile
generator
str.join
torch.FloatTensor
pointcloud_inpainting.max
tuple
objZip.read.decode
numpy.linspace
numpy.float32.objMat.transpose.astype.numpy.ascontiguousarray.torch.FloatTensor.cuda
process_autozoom
objInpainted.view.expand
spatial_filter.float
netMaskrcnn
disparity_estimation
npyReference.numpy.log10.npyEstimate.numpy.log10.__abs__.mean.item
torch.nn.MaxPool2d
cupy.cuda.compile_with_cache
tenExisting.float
process_kenburns
intMask.objPredictions.item
re.search
torchvision.models.vgg19_bn.features.eval
launch_kernel
strScript.replace.replace
fltRmse.append
preprocess_kernel
compute_directed_depth_error
depth_to_points
tenMerge.tenMask.sum.item
intMerge.intMerge.objPredictions.float
Disparity.cuda
cv2.imwrite
flask.Flask.send_static_file
cv2.getRectSubPix
print
super.__init__
tenPreprocessed.device.tenPreprocessed.dtype.torch.tensor.view
Inpaint.cuda
npyImage.copy.transpose.astype
os.path.abspath
int
intMerge.objPredictions.item
tenInput.clone.contiguous
type
tenDepth.data_ptr
npyImage.transpose
disparity_refinement
tenDepth.detach.cpu
objInpainted.max.objInpainted.spatial_filter.abs
intArg.strArgs.replace.replace
spatial_filter
intColumn.str.intRow.str.intColumn.str.intRow.str.self._modules
self.netSemantics
tenInput.new_zeros
tenMerge.tenMask.clip.sum
numpy.float32.npyImage.copy.transpose.astype.numpy.ascontiguousarray.torch.FloatTensor.cuda
tenDisparity.min.item
numpy.log10
self.netMain
fltY.fltX.torch.FloatTensor.view.cuda
objMat.transpose.astype
scipy.io.loadmat.transpose
compute_planarity_error
torchvision.models.vgg19_bn
netSemantics
pe_ori.extend
tenRender.detach.cpu
tenImage.clone.clone
intTest.npyImages.transpose.copy
intMask.intMask.objPredictions.float
pointcloud_inpainting
self.netDisparityOne
gevent.pywsgi.WSGIServer
torch.cat.new_ones
open.read
tenDisparity.max.item
compute_depth_boundary_error
cupy.int32
npyReference.npyReference.npyEstimate.__abs__.mean.item
intTest.npyDepths.transpose.flatten
time.time.str.split
npyEstimate.npyReference.npyReference.npyEstimate.numpy.maximum.mean.item
tenDepth.detach
range
torch.is_tensor
pe_fla_floor.tolist
tenPlane.neg.torch.nn.functional.max_pool2d.neg.sum
tenDepth.cpu.numpy.flatten
len
self.add_module
torch.nn.functional.pad
tenRender.detach.cpu.numpy
tenInput.flip
self.netInput
random.choice
re.search.group
tenOutput.unfold.unfold
tenDisparity.shape.tenDisparity.view.mean.view
reversed
intArg.strArgs.replace
fltBaseline.npyDisparity.clip.astype
tenImage.clone.view
tenMerge.sum.item
sorted
torch.nn.functional.interpolate
Semantics.cuda.eval
shutil.rmtree
intArg.strArgs.replace.replace.strip
strKey.replace
tenDisparity.max.min
npyReference.npyReference.npyEstimate.__pow__.mean.item
npyResult.reversed.list.npyResult.npyFrame.npyFrame.moviepy.editor.ImageSequenceClip.write_videofile
fltShiftZ.fltShiftY.fltShiftX.torch.FloatTensor.view.cuda
self.netDisparityFou
flask.send_file
list
npyReference.npyEstimate.__pow__
tenDepth.cpu
tenImage.shape.tenImage.view.std
torch.nn.Conv2d
npyReference.npyReference.npyEstimate.__pow__.mean
fltBaseline.npyDisparity.clip
tenRender.detach.cpu.numpy.transpose.clip.astype
Refine.cuda
fltShiftZ.fltShiftY.fltShiftX.torch.FloatTensor.view
strTensor.objVariables.stride
tenImage.shape.tenImage.view.mean.view
numpy.ones
scipy.io.loadmat
self.netImage.clone
Basic
tenPlane.sum.flatten
tenZee.tenData.tenInput.fltBaseline.fltFocal.intHeight.intWidth.preprocess_kernel.launch_kernel
fltThr1.append
npyOutputs.append
numpy.ascontiguousarray
self.netImageOne
tenDisparity.clone.clone
cv2.imread.transpose
self.netRefine
tenContext.tenDisparity.tenImage.torch.cat.view
torch.nn.PReLU
intArg.intSizes.item
tenPlane.neg.torch.nn.functional.max_pool2d.neg.neg
Inpaint
pointcloud_inpainting.view
strKernel.replace.replace
tenInput.clone.median
open
pe_ori_table.tolist
tenDepth.shape.tenDepth.shape.tenDepth.shape.torch.linspace.view.expand
Refine
flask.Flask.route
torch.hub.download_url_to_file
Refine.cuda.eval.load_state_dict
tenInput.clone.detach
tenInput.clone.unfold
tenPlane.neg.torch.nn.functional.max_pool2d.neg
torch.tensor
base64.b64decode
torch.nn.functional.conv2d
valid.sum
tenExisting.float.sum.item
npyEstimate.npyReference.npyReference.npyEstimate.numpy.maximum.mean
torch.set_grad_enabled
valid.pred.flatten
process_inpaint
self.netDisparity.clone
cv2.IMWRITE_JPEG_QUALITY.numpy.uint8.numpy.ones.cv2.imencode.tobytes
tenDisparity.max.max
self.netDisparityTwo
torch.cat
objSettings.clone
torch.nn.functional.interpolate.cpu
h5py.File.close
self.netDisparity.max
flask.request.form.split
intArg.intStrides.item
numpy.linspace.tolist
depth_to_points.view
sum
tenRender.detach.cpu.numpy.transpose
tenMask.expand.objInpainted.view.view
torch.linspace
cv2.minMaxLoc
objZip.namelist
netDisparity
objZip.read
tenPlane.sum.item
tenOutput.tenZee.tenData.tenInput.fltBaseline.fltFocal.intHeight.intWidth.preprocess_kernel.launch_kernel
tenDisparity.shape.tenDisparity.view.mean
objMatch.group.split
Disparity.cuda.eval.load_state_dict
torchvision.models.detection.maskrcnn_resnet50_fpn.cuda
os.getpid
self.netVgg
tenOutput.contiguous.view
tenPoints.view.view
</pre>
</details>

@developer
Could please help me check this issue?
May I pull a request to fix it?
Thank you very much.

**Comment by sniklaus** [comment]
My apologies, but I am not interested in interacting with a bot.
